src/crypto/convert.js

```diff
@@ -24,7 +24,18 @@ export const bufferToBigNumberString = bigNumberBuffer =>
 
 export const bufferToHex = buffer => naclInstance.to_hex(buffer);
 
-export const hexToBuffer = hex => Buffer.from(hex, 'hex');
+const hexRegex = /^([0-9]|[a-f])+/gi;
+export const hexToBuffer = hex => {
+	if (typeof hex !== 'string') {
+		throw new TypeError('argument must be string');
+	}
+	const matchedHex = hex.match(hexRegex) || [];
+	if (matchedHex.length === 0) {
```

Comment:
Why not just this?
```
const matchedHex = hex.match(hexRegex);
if (!matchedHex) {
    return Buffer.alloc(0);
}

Reply:
In the following code, it uses first element like `matchedHex[0]`, so I think it would be better to check length here already